tests/integration/helpers/cluster.py

```diff
@@ -392,11 +394,13 @@ def __init__(self, base_path, name=None, base_config_dir=None, server_bin_path=N
     def cleanup(self):
         # Just in case kill unstopped containers from previous launch
         try:
-            result = run_and_check(f'docker container list --all --filter name={self.project_name} | wc -l', shell=True)
+            # We need to have "^/" and "$" in the "--filter name" option below to filter by exact name of the container, see
+            # https://stackoverflow.com/questions/48767760/how-to-make-docker-container-ls-f-name-filter-by-exact-name
+            result = run_and_check(f'docker container list --all --filter name=^/{self.project_name}$ | wc -l', shell=True)
             if int(result) > 1:
-                logging.debug(f"Trying to kill unstopped containers for project{self.project_name}...")
-                run_and_check(f'docker kill $(docker container list --all --quiet --filter name={self.project_name})', shell=True)
```

Comment:
Without my fix a successfully finished test in `test_dictionaries_all_layouts_separate_sources/test_http.py` was able to stop both the containers `projectroottestdictionariesalllayoutsseparatesourceshttp` and `projectroottestdictionariesalllayoutsseparatesourceshttps`, which could cause a parallel running test in `test_dictionaries_all_layouts_separate_sources/test_https.py` to fail.